### PR TITLE
chore: migrate to logging from @actions/core

### PR DESCRIPTION
**Pull request by varl**
Use logging from `@actions/core`[1] over console-based logging recommended by the javascript action template[2].

[1]: https://github.com/actions/toolkit/tree/main/packages/core#logging
[2]: https://github.com/actions/javascript-action/blob/main/index.js

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -100,7 +100,7 @@ const getOptionalInput = (name: string) => core.getInput(name) || undefined;
       return;
     }
     case hasChangesets && !hasNonEmptyChangesets:
-      console.log("All changesets are empty; not creating PR");
+      core.warning("All changesets are empty; not creating PR");
```
Comment (Andarist) [inline src/index.ts:103]:
```suggestion
      core.info("All changesets are empty; not creating PR");
```

Comment (Andarist) [reply src/index.ts:103]: This is a "normal situation" so I think that it shouldn't be a warning. Happy to discuss this further if you feel differently though.

File: .changeset/blue-hornets-marry.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+"@changesets/action": patch
+---
+
+migrate to logging provided by @actions/core
```
Comment (Andarist) [inline .changeset/blue-hornets-marry.md:5]:
```suggestion
Use logging provided by `@actions/core`
```

File: src/index.ts (hunk 1 above)
Comment (varl) [reply src/index.ts]: No, that makes sense, thanks for the fix.

**Pull request merged**